wordpress-mobile/WordPress-iOS
# UL&S: PasswordViewController: send magic link

Ref. #14009
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/403/

This PR:
- adds the ability to send a magic link on the password entry screen
- updates the button color in Unified to "hero blue" instead of .brand
- (doesn't have Tracks set up for the magic link event)

### To test
1. Enable the `.unifiedLoginLink` feature flag
3. Switch to local pod development in the podfile
2. Ensure the Auth PR is checked out
3. In the Auth PR, visit the `LoginPrololgueViewController`. Find function `presentUnifiedLoginMagicLinkView()` and change the first line to `guard let toVC = PasswordViewController.instantiate(from: .password) else`
4. `rake dependencies`
5. Build and run
6. Log In > Continue with WordPress.com
7. Notice the new button in the PasswordViewController.
8. Select the "Continue" button without entering a password. (The email address will disappear, since this is a hack-y way to show the screen.) An error message will appear.
9. Select "Get a login link by email". An error alert will pop up.
10. Optional: test a valid address. Go to `PasswordViewController` `viewDidLoad()` and change the `loginFields.username` string. It will navigate you to the "magic link sent!" view (but will display the wrong email address because of the hack-y bypass).

### Screenshots

| Design | Screenshot |
| --- | --- |
| <img width="344" alt="Screen Shot 2020-08-24 at 8 12 28 AM" src="https://user-images.githubusercontent.com/1062444/91182398-bd6e2900-e6af-11ea-8447-206884d5bb01.png"> | <a href="https://user-images.githubusercontent.com/1062444/91183386-0672ad00-e6b1-11ea-82d1-13e50029122d.png"><img src="https://user-images.githubusercontent.com/1062444/91183386-0672ad00-e6b1-11ea-82d1-13e50029122d.png" width="350" /></a> |

| Password errors are in-line | Magic link errors are alerts |
| --- | --- |
| <a href="https://user-images.githubusercontent.com/1062444/91183426-15f1f600-e6b1-11ea-805d-6d8038164a7a.png"><img src="https://user-images.githubusercontent.com/1062444/91183426-15f1f600-e6b1-11ea-805d-6d8038164a7a.png" width="350" /></a> | <a href="https://user-images.githubusercontent.com/1062444/91183459-20ac8b00-e6b1-11ea-9667-f29e1337adbd.png"><img src="https://user-images.githubusercontent.com/1062444/91183459-20ac8b00-e6b1-11ea-9667-f29e1337adbd.png" width="350" /></a> |

PR submission checklist:
